recipes/libigl/2.x.x/conanfile.py

```diff
@@ -34,106 +32,140 @@ def _minimum_cpp_standard(self):
     def _minimum_compilers_version(self):
         return {
             "Visual Studio": "16",
+            "msvc": "1920",
             "gcc": "6",
             "clang": "3.4",
             "apple-clang": "5.1",
         }
 
-    def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._minimum_cpp_standard)
-        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
-        if not min_version:
-            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
-                self.name, self.settings.compiler))
-        else:
-            if tools.Version(self.settings.compiler.version) < min_version:
-                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
-                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
-        if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime and not self.options.header_only:
-            raise ConanInvalidConfiguration("Visual Studio build with MT runtime is not supported")
-        if "arm" in self.settings.arch or "x86" is self.settings.arch:
-            raise ConanInvalidConfiguration("Not available for arm. Requested arch: {}".format(self.settings.arch))
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.header_only:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        if not self.options.header_only:
+            cmake_layout(self, src_folder="src")
 
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+    def requirements(self):
+        self.requires("eigen/3.4.0")
+        if Version(self.version) >= "2.4.0":
+            self.requires("gmp/6.2.1")
+            self.requires("mpfr/4.1.0")
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
```

Comment:
```suggestion
        if self.info.settings.get_safe("compiler.cppstd"):
```
single package ID does not provide a compiler on validate()